src/main/java/com/cloudbees/jenkins/plugins/bitbucket/BitbucketCredentials.java

```diff
@@ -47,11 +47,28 @@
  * Utility class for common code accessing credentials
  */
 class BitbucketCredentials {
-    private BitbucketCredentials() {
-        throw new IllegalAccessError("Utility class");
+
+    private final String serverUrl;
+    private final SCMSourceOwner context;
```

Comment:
As you noted, it may not be safe to hold a reference to this.  I don't know either.

Reply:
I was hoping you would not notice :-D

Reply:
I triied to dig into it, but I cannot quite figure out where SCMSourceOwner is coming from and if it is safe to hold a reference to it. But to get credentials we need an "Item" to get credentials in reference to and I am not sure how else to get it... :-( I am a bit stuck on this one